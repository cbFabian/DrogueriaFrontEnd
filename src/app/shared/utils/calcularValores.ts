export const calcularValores = (producto: string, 
    reunion: string, esFestivo: boolean) => {
        if(!producto && !reunion){
            return 0;
        }
        if(!producto){
            return JSON.parse(reunion).precio;
        }
        if(!reunion){
            return JSON.parse(producto).precio;
        }
        let valorTotal: number = JSON.parse(producto).precio + JSON.parse(reunion).precio;
        if(esFestivo){
            valorTotal *= 2;
        }                
        return valorTotal;
};
