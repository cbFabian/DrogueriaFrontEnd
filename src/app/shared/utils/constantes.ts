export const constantes = {
    HORAS24: 24,
    HORA15: 15,
    HORA19: 19,
    MINUTOS60: 60,
    SEGUNDOS60: 60,
    MILISEGUNDOS: 1000,
    POSICION_DOS: 2,
    POSICION_TRES: 3,
    POSICION_CUATRO: 4,
    NUMERO_DOS: 2,
    NUMERO_TRES: 3,
    HORAMINIMA: 4,
    NUMERO_CINCO: 5,
    HORAMAXIMA: 8,
};
