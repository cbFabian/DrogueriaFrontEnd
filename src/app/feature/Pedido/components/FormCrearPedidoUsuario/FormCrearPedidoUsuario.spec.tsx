import * as React from 'react';
import { RenderResult, fireEvent, render, wait } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { FormCrearPedidoUsuario } from '.';
import { setTextEvent } from 'app/shared/utils/test';

const NUMERO_DOS = 2;


describe('FormCrearPedidoUsuario test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof FormCrearPedidoUsuario> & {
    onSubmit: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      formTitle: 'Form test',
      onSubmit: stub(),
      validarDiaFestivo: stub(),
      productos: [{
                nombre: 'Paella Española',
                precio: 40000,
                detalle: 'Verduras y sustituye'
            }
        ],
      usuarioPedido: {nombre: 'Lorem', clave: '1234'},
      reuniones:[{
                tipo: 'TIPO_PEQUENA',
                precio: 25000
            }, {
                tipo: 'TIPO_MEDIANA',
                precio: 50000
            }, {
                tipo: 'TIPO_GRANDE',
                precio: 75000
            }
        ],
        mensajePedido: '',
        mensajeExcepcion: '',
        esFestivo: false,
    };
    componentWrapper = render(<FormCrearPedidoUsuario {...componentProps} />);
  });

  it('should match snapshot of crear pedidos', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });

  it('should fail on submit two fields missing on crear pedidos', async () => {
    const elem = componentWrapper.container;
    const horasDeServicio = elem.querySelector('input[name="horasDeServicio"]');
    const producto = elem.querySelector('input[name="producto"]');
    const reunion = elem.querySelector('input[name="reunion"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
        horasDeServicio && fireEvent.change(horasDeServicio, setTextEvent('horasDeServicio', ''));
    });
    await wait(() => {
      producto && fireEvent.change(producto, setTextEvent('producto', `"{\\"nombre\\":\\"Paella Española\\",
        \\"precio\\":\\"40000\\",\\"detalle\\":\\"Verduras y sustituye\\"}"`));
    });
    await wait(() => {
      reunion && fireEvent.change(reunion, setTextEvent('reunion', `"{\\"tipo\\":\\"TIPO_PEQUENA\\",\\"precio\\":25000}"`));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');

    expect(spans.length).toBe(NUMERO_DOS);
    expect(spans[0].textContent).toBe('El campo Direccion es requerido.');
    expect(spans[1].textContent).toBe('El campo Horas de Servicio es requerido.');
  });

  it('should fail on submit two fields missing on crear pedidos', async () => {
    const elem = componentWrapper.container;
    const producto = elem.querySelector('select[name="producto"]');
    const reunion = elem.querySelector('select[name="reunion"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
        producto && fireEvent.change(producto, setTextEvent('producto', `"{\\"nombre\\":\\"Paella Española\\",
          \\"precio\\":\\"40000\\",\\"detalle\\":\\"Verduras y sustituye\\"}"`));
    });
    await wait(() => {
      reunion && fireEvent.change(reunion, setTextEvent('reunion', `"{\\"tipo\\":\\"TIPO_PEQUENA\\",\\"precio\\":25000}"`));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(NUMERO_DOS);
    expect(spans[0].textContent).toBe('El campo Direccion es requerido.');
    expect(spans[1].textContent).toBe('Minimo 4 horas');
  });

  it('should fail on submit two fields missing on crear pedidos', async () => {
    const elem = componentWrapper.container;

    const producto = elem.querySelector('select[name="producto"]');
    const reunion = elem.querySelector('select[name="reunion"]');
    const fechaRealizacion = elem.querySelector('input[name="fechaRealizacion"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
        producto && fireEvent.change(producto, setTextEvent('producto', `"{\\"nombre\\":\\"Paella Española\\",
          \\"precio\\":\\"40000\\",\\"detalle\\":\\"Verduras y sustituye\\"}"`));
    });
    await wait(() => {
        reunion && fireEvent.change(reunion, setTextEvent('reunion', `"{\\"tipo\\":\\"TIPO_PEQUENA\\",\\"precio\\":25000}"`));
    });
    await wait(() => {
        fechaRealizacion && fireEvent.change(fechaRealizacion, setTextEvent('fechaRealizacion', 'January 4, 2022 3:00 PM'));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(NUMERO_DOS);
    expect(spans[0].textContent).toBe('El campo Direccion es requerido.');
    expect(spans[1].textContent).toBe('Minimo 4 horas');
  });

  it('should fail on submit one fields missing on crear pedidos', async () => {
    const elem = componentWrapper.container;

    const producto = elem.querySelector('select[name="producto"]');
    const reunion = elem.querySelector('select[name="reunion"]');
    const fechaRealizacion = elem.querySelector('input[name="fechaRealizacion"]');
    const direccion = elem.querySelector('input[name="direccion"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
        producto && fireEvent.change(producto, setTextEvent('producto', `"{\\"nombre\\":\\"Paella Española\\",
          \\"precio\\":\\"40000\\",\\"detalle\\":\\"Verduras y sustituye\\"}"`));
    });
    await wait(() => {
        reunion && fireEvent.change(reunion, setTextEvent('reunion', `"{\\"tipo\\":\\"TIPO_PEQUENA\\",\\"precio\\":25000}"`));
    });
    await wait(() => {
        fechaRealizacion && fireEvent.change(fechaRealizacion, setTextEvent('fechaRealizacion', 'January 4, 2022 3:00 PM'));
    });
    await wait(() => {
        direccion && fireEvent.change(direccion, setTextEvent('direccion', 'calle 10 # 30-40'));
    });

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(1);
    expect(spans[0].textContent).toBe('Minimo 4 horas');
  });
});
