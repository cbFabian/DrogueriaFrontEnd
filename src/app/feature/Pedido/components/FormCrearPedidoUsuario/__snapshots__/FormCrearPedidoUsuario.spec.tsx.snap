// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCrearPedidoUsuario test should match snapshot of crear pedidos 1`] = `
<div>
  <form>
    <h2>
      Form test
    </h2>
    <label>
      Productos Disponibles:
       
    </label>
    <select
      class="sc-AxhCb kUujfn"
      name="producto"
    >
      <option
        value="{\\"nombre\\":\\"Paella Española\\",\\"precio\\":40000,\\"detalle\\":\\"Verduras y sustituye\\"}"
      >
        Paella Española
      </option>
    </select>
    <label>
      Tipos de Reuniones:
       
    </label>
    <select
      class="sc-AxhCb kUujfn"
      name="reunion"
    >
      <option
        value="{\\"tipo\\":\\"TIPO_PEQUENA\\",\\"precio\\":25000}"
      >
        TIPO_PEQUENA
      </option>
      <option
        value="{\\"tipo\\":\\"TIPO_MEDIANA\\",\\"precio\\":50000}"
      >
        TIPO_MEDIANA
      </option>
      <option
        value="{\\"tipo\\":\\"TIPO_GRANDE\\",\\"precio\\":75000}"
      >
        TIPO_GRANDE
      </option>
    </select>
    <label>
      Fecha y hora de inicio del evento:
    </label>
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <input
          class=""
          name="fechaRealizacion"
          type="text"
          value="January 21, 2022 3:00 PM"
        />
      </div>
    </div>
    <input
      class="sc-AxirZ eAukyj"
      name="fechaRealizacionAux"
      type="hidden"
      value="Fri Jan 21 2022 15:00:00 GMT-0500 (Colombia Standard Time)"
    />
    <input
      class="sc-AxirZ eAukyj"
      name="direccion"
      placeholder="Ingrese la Dirección del evento. (Solo en Medellín)"
      value=""
    />
    <input
      class="sc-AxirZ eAukyj"
      name="horasDeServicio"
      placeholder="Ingrese el numero Horas del servicio, (minimo 4 horas)"
      type="number"
      value="0"
    />
    <label
      for="valorTotalPedido"
    >
      Valor total de su pedido:
      65000
    </label>
    <button
      class="sc-AxjAm kgxTKy"
      type="submit"
    >
      Hacer Pedido
    </button>
  </form>
</div>
`;
